frontend/src/pages/UserPreferencesPage.tsx

```diff
@@ -1,23 +1,75 @@
-import { IonButtons, IonContent, IonHeader, IonImg, IonMenuButton, IonPage, IonTitle, IonToolbar } from "@ionic/react";
+import { IonButtons, IonContent, IonHeader, IonImg, IonLabel, IonMenuButton, IonPage, IonTitle, IonToggle, IonToolbar } from "@ionic/react";
+import { useEffect } from "react";
 import logo from "../../public/images/OeffisLogo1.svg";
+import { PersistenceService } from "../services/persistence/PersistenceService";
+import styles from "./UserPreferencesPage.module.css";
 
-const UserPreferencesPage: React.FC = () => (
-  <IonPage id="main-content">
-    <IonHeader>
-      <IonToolbar>
-        <IonButtons slot="start">
-          <IonMenuButton />
-        </IonButtons>
-        <div className="menuBar">
-          <IonTitle>Öffis</IonTitle>
-          <IonImg className="menuLogo" src={logo} />
+export enum UserPreferences {
+  isDarkThemeEnabled = "isDarkThemeEnabeld"
+}
+export enum UserPreferencesValues {
+  enabled = "enabled",
+  disabled = "disabled",
+}
+
+export interface UserPreferencesPageProps {
+  isDarkThemeEnabled: boolean,
+  setIsDarkThemeEnabled: (value: boolean) => void
+}
+
+const UserPreferencesPage: React.FC<UserPreferencesPageProps> = (props) => {
+
+  const persistance = new PersistenceService();
+
+  const setIsDarkThemeEnabledPersistence = (): void => {
+    props.setIsDarkThemeEnabled(!props.isDarkThemeEnabled);
+    persistance.set(
+      UserPreferences.isDarkThemeEnabled,
+      props.isDarkThemeEnabled
+        ? UserPreferencesValues.disabled
+        : UserPreferencesValues.enabled
+    );
+  };
+
+  useEffect(() => {
+    persistance.get(UserPreferences.isDarkThemeEnabled) !== undefined
+      ? persistance.get(UserPreferences.isDarkThemeEnabled) === UserPreferencesValues.enabled
+        ? props.setIsDarkThemeEnabled(true)
+        : props.setIsDarkThemeEnabled(false)
+      : props.setIsDarkThemeEnabled(false);
+    props.isDarkThemeEnabled
+      ? document.body.setAttribute("color-theme", "dark")
+      : document.body.removeAttribute("color-theme");
+  }, [props.isDarkThemeEnabled]);
+
+  return (
+    <IonPage id="main-content">
+      <IonHeader>
```

Comment:
Please don't use the IonHeader to make your own header. There is a Header Component for consistency.

Reply:
Header component is used instead.